File: frontend/src/api/auditCleanup.ts
/**
 * API client para gerenciamento de logs de auditoria e limpeza
 */

import api from './axios';

export interface AuditStats {
  total_logs: number;
  oldest_log_date: string | null;
  newest_log_date: string | null;
  logs_by_action: Record<string, number>;
  logs_by_month: Record<string, number>;
  size_estimation_mb: number;
}

export interface CleanupResult {
  success: boolean;
  message: string;
  deleted_count: number;
  freed_space_mb: number;
  oldest_remaining_date: string | null;
  cleanup_date: string;
}

export interface AutoCleanupResult {
  success: boolean;
  message: string;
  cleanup_result: CleanupResult;
}

class AuditCleanupAPI {
  /**
   * Obter estatísticas dos logs de auditoria
   */
  async getStats(): Promise<{ success: boolean; stats: AuditStats }> {
    const response = await api.get('/audit-cleanup/stats');
    return response.data;
  }

  /**
   * Executar limpeza manual de logs antigos
   */
  async cleanupOldLogs(monthsToKeep: number = 6): Promise<CleanupResult> {
    const response = await api.post('/audit-cleanup/cleanup', null, {
      params: { months_to_keep: monthsToKeep }
    });
    return response.data;
  }

  /**
   * Configurar limpeza automática
   */
  async enableAutoCleanup(monthsToKeep: number = 6): Promise<AutoCleanupResult> {
    const response = await api.post('/audit-cleanup/auto-cleanup', null, {
      params: { months_to_keep: monthsToKeep }
    });
    return response.data;
  }
}

export const auditCleanupApi = new AuditCleanupAPI();

File: backend/app/routers/audit_cleanup.py
"""
Router para gerenciamento de logs de auditoria e limpeza automática
"""
from fastapi import APIRouter, Depends, HTTPException
from app.core.deps import get_current_user, is_admin
from app.models.user import User
from app.services.audit_cleanup import AuditLogCleanupService
from typing import Optional

router = APIRouter()
cleanup_service = AuditLogCleanupService()

@router.get("/stats")
async def get_audit_stats(
    current_user: User = Depends(is_admin)
):
    """
    Retorna estatísticas detalhadas dos logs de auditoria
    """
    try:
        stats = await cleanup_service.get_audit_stats()
        return {
            "success": True,
            "stats": stats
        }
    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"Erro ao obter estatísticas: {str(e)}"
        )

@router.post("/cleanup")
async def cleanup_old_logs(
    months_to_keep: int = 6,
    current_user: User = Depends(is_admin)
):
    """
    Remove logs de auditoria mais antigos que o período especificado
    Padrão: 6 meses
    """
    try:
        if months_to_keep < 1 or months_to_keep > 24:
            raise HTTPException(
                status_code=400,
                detail="Período deve estar entre 1 e 24 meses"
            )
        
        result = await cleanup_service.cleanup_old_logs(months_to_keep)
        return result
        
    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"Erro na limpeza de logs: {str(e)}"
        )

@router.post("/auto-cleanup")
async def enable_auto_cleanup(
    months_to_keep: int = 6,
    current_user: User = Depends(is_admin)
):
    """
    Configura limpeza automática dos logs de auditoria
    (Esta rota seria expandida futuramente para configurar cron jobs)
    """
    try:
        # Por enquanto, apenas executa a limpeza uma vez
        # No futuro, isso poderia configurar um job agendado
        result = await cleanup_service.cleanup_old_logs(months_to_keep)
        
        return {
            "success": True,
            "message": f"Limpeza executada - configuração para {months_to_keep} meses",
            "cleanup_result": result
        }
        
    except Exception as e:
        raise HTTPException(
            status_code=500,
            detail=f"Erro na configuração de limpeza automática: {str(e)}"
        )
